app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java

```diff
@@ -97,10 +62,18 @@ private NearbyPlaces(){
         return places;
     }
 
-    private List<Place> getFromWikidataQuery(LatLng cur, String lang, double radius)
+    private List<Place> getFromWikidataQuery(Context context,
+                                             LatLng cur,
+                                             String lang,
+                                             double radius)
             throws IOException {
         List<Place> places = new ArrayList<>();
-        String query = WIKIDATA_QUERY_TEMPLATE.replace("${RADIUS}", "" + radius)
+
+        String query = FileUtils.readFromFile(context, "queries/nearby_query.txt");
+
```

Comment:
I am currently not working on it. It would be great if you could complete the thing and create a PR.